variables.tf

```diff
@@ -170,4 +191,16 @@ variable "verify_worker_network_readiness" {
   default     = true
 }
 
+variable "addons" {
+  type = object({
+    cluster-autoscaler   = string
+    vpc-block-csi-driver = string
+  })
+  description = "Flag indicating whether or not to enable autoscaling."
+  default = {
+    cluster-autoscaler   = "1.0.8"
+    vpc-block-csi-driver = "5.0"
+  }
+}
```

Comment:
I think this will result in change requests/issues.

Neither are optional, so you have to set both; plus this excludes all other possible addons that a consumer may want to specify.

Also I think the logic for using this variable is `if public and ignore_worker_count` then install addons means that both addons will be installed for every consumer that is ignoring the worker count.

Maybe a better solution is to cover all known addons, to make them all optional, with vpc-block-csi-driver the only one having a default value in the `type` definition and then set the default value for the block to null.
Then change the logic for ibm_container_addons to count on addons != null (replacing the ignore_worker_pool_size_changes piece)

The count on `kubernetes_config_map_v1_data` may need updating as well.